@nomiclabs/hardhat-solhint does not work

### Version of Hardhat

2.12.7

### What happened?

```
NomicLabsHardhatPluginError: An error occurred loading the solhint formatter stylish
    at getFormatter (.../node_modules/@nomiclabs/hardhat-solhint/src/index.ts:21:11
```

`Caused by: Error: Cannot find module 'eslint/lib/formatters/stylish'`

calling:

https://github.com/NomicFoundation/hardhat/blob/main/packages/hardhat-solhint/src/index.ts#L12


### Minimal reproduction steps

- create hardhat project using typescript and default settings
- add hardhat-solhint and enable in config
- run npx hardhat check
- 

### Search terms

solhint hardhat-solhint

--- a/.changeset/strange-drinks-cry.md
+++ b/.changeset/strange-drinks-cry.md
@@ -1,0 +1,5 @@
+---
+"@nomiclabs/hardhat-solhint": patch
+---
+
+Update solhint to v3.4.0 and fix error with ESLint formatter not found.

--- a/packages/hardhat-solhint/package.json
+++ b/packages/hardhat-solhint/package.json
@@ -34,7 +34,7 @@
     "README.md"
   ],
   "dependencies": {
-    "solhint": "^3.0.0"
+    "solhint": "^3.4.0"
   },
   "devDependencies": {
     "@types/chai": "^4.2.0",
@@ -54,6 +54,7 @@
     "mocha": "^10.0.0",
     "prettier": "2.4.1",
     "rimraf": "^3.0.2",
+    "sinon": "^9.0.0",
     "ts-node": "^10.8.0",
     "typescript": "~4.7.4"
   },

--- a/packages/hardhat-solhint/src/index.ts
+++ b/packages/hardhat-solhint/src/index.ts
@@ -13,7 +13,7 @@ function getFormatter(formatterName = "stylish") {
   try {
     const solhintPath = require.resolve("solhint");
     const formatterPath = require.resolve(
-      `eslint/lib/formatters/${formatterName}`,
+      `solhint/lib/formatters/${formatterName}`,
       { paths: [solhintPath] }
     );
     return require(formatterPath);

--- a/packages/hardhat-solhint/src/index.ts
+++ b/packages/hardhat-solhint/src/index.ts
@@ -11,10 +11,8 @@ function getDefaultConfig() {
 
 function getFormatter(formatterName = "stylish") {
   try {
-    const solhintPath = require.resolve("solhint");
     const formatterPath = require.resolve(
-      `solhint/lib/formatters/${formatterName}`,
-      { paths: [solhintPath] }
+      `solhint/lib/formatters/${formatterName}`
     );
     return require(formatterPath);
   } catch (ex: any) {

--- a/packages/hardhat-solhint/src/index.ts
+++ b/packages/hardhat-solhint/src/index.ts
@@ -84,7 +84,7 @@ function printReport(reports: any) {
 subtask("hardhat-solhint:run-solhint", async (_, { config }) => {
   const { processPath } = require("solhint/lib/index");
   return processPath(
-    join(config.paths.sources, "**", "*.sol"),
+    join(config.paths.sources, "**", "*.sol").replace(/\\/g, "/"),
     await getSolhintConfig(config.paths.root)
   );
 });

--- a/packages/hardhat-solhint/test/tests.ts
+++ b/packages/hardhat-solhint/test/tests.ts
@@ -1,5 +1,6 @@
 import { assert } from "chai";
 import { unlink, writeFile, writeJson } from "fs-extra";
+import sinon from "sinon";
 
 import { useEnvironment } from "./helpers";
 
@@ -33,6 +34,13 @@ describe("Solhint plugin", function () {
         // and it was failing very often when solhint released new versions
         reports[0].reports.length >= 5
       );
+    });
+
+    it("should run the check task without throwing an error", async function () {
+      const consoleLogStub = sinon.stub(console, "log");
+      await this.env.run("check");
+      assert.isTrue(consoleLogStub.calledOnce);
+      consoleLogStub.restore();
     });
   });
 

